### PR TITLE
[NAN-751] Add migrate command to make migration easier

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Adds a cli migration command `nango migrate-to-directories` to easily change to the nested cli directory structure

## Issue ticket number and link
NAN-751

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/cli/lib/services/migration.service.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        }
+        return;
+    }
+    try {
+        await fs.promises.mkdir(dirPath, { recursive: true });
+        if (debug) {
+            printDebug(`Created directory at ${dirPath}.`);
+        }
+    } catch (error) {
+        console.error(chalk.red(`There was an issue creating the directory at ${dirPath}.`), error);
```
Comment (bodinsamuel) [inline packages/cli/lib/services/migration.service.ts]: this seems to be a critical operation and if it fails everything else will fail. imo it should not be try/catch or stop the processing

File: packages/cli/lib/services/migration.service.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+async function updateModelImport(filePath: string, debug = false): Promise<void> {
+    try {
+        const data = await fs.promises.readFile(filePath, 'utf8');
+        const updatedData = data.replace(/from '\.\/models/, "from '../../models");
+        await fs.promises.writeFile(filePath, updatedData, 'utf8');
+        if (debug) {
+            printDebug(`Updated imports in ${filePath}.`);
+        }
+    } catch (error) {
+        console.error(chalk.red(`There was an issue updating the imports in ${filePath}.`), error);
```
Comment (bodinsamuel) [inline packages/cli/lib/services/migration.service.ts:74]: same

File: packages/cli/lib/services/migration.service.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    }
+    try {
+        await fs.promises.rename(source, destination);
+        if (debug) {
+            printDebug(`Moved file from ${source} to ${destination}.`);
+        }
+
+        return true;
+    } catch (error) {
+        console.error(chalk.red(`There was an issue moving the file from ${source} to ${destination}.`), error);
```
Comment (bodinsamuel) [inline packages/cli/lib/services/migration.service.ts]: same

**Review by bodinsamuel — APPROVED** [review]
Works great locally 👌🏻 
Maybe a safer way would be to copy file and only delete the original at the end

File: packages/cli/lib/services/migration.service.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/cli/lib/services/migration.service.ts]: Updated.

File: packages/cli/lib/services/migration.service.ts (hunk 3 above)
Comment (khaliqgant) [reply packages/cli/lib/services/migration.service.ts]: Updated.

File: packages/cli/lib/services/migration.service.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/cli/lib/services/migration.service.ts:74]: This isn't as critical so left this in the try, catch

**Pull request merged**